command/daemon/command.go

```diff
@@ -77,21 +83,23 @@ func (c *command) CobraCommand() *cobra.Command {
 }
 
 func (c *command) Execute(cmd *cobra.Command, args []string) {
+	v := c.serverFactory().Config().Viper
+
 	// Merge the given command line flags with the given environment variables and
-	// the given config file, if any. The merged flags will be applied to the
-	// global Flags struct.
-	err := MergeFlags(cmd.Flags())
+	// the given config files, if any. The merged flags will be applied to the
+	// given viper.
+	err := microflag.Merge(v, cmd.Flags(), v.GetStringSlice(Flag.Config.Dirs), v.GetStringSlice(Flag.Config.Files))
```

Comment:
This is how clients can access configuration. 

```
dirst := v.GetStringSlice(Flag.Config.Dirs)
```

The cool thing is that only the viper has to be propagated through the microservice's service packages.